extensions/ql-vscode/src/sarif-utils.ts

```diff
@@ -72,7 +72,18 @@ export function getPathRelativeToSourceLocationPrefix(
   sourceLocationPrefix: string,
   sarifRelativeUri: string
 ) {
-  const normalizedSourceLocationPrefix = sourceLocationPrefix.replace(/\\/g, '/');
+  // convert a platform specific path into encoded path uri segments
+  // need to be careful about drive letters
+  let prefix = '';
+  if (sourceLocationPrefix[1] === ':') {
```

Comment:
This ins't correct as windows paths should have a `/` before the drive letter. 
I think the following is correct (it is what we do in java and seems to mostly match vscode-uri). I also notice that vscode-uri is availiable in npm standalone. Is it possible to use it directly and therefore allow it to be used in tests. Then we avoid some of this awkwardness of implementing uri operations standalone.

```
const normalisedPath = path.replace('/\\/g', '/');
const pathComponent = path[0] ==  '/' ? path : "/" + path;
const encodedPathComponent = pathComponent .split('/').map(encodeURIComponent).join('/');
return `file:${encodedPathComponent}/${sarifRelativeUri}`
```

Reply:
OK. I see what you're saying. I think we've been lucky that the URI implementation is flexible enough to handle these uris. I can change this.